unstructured/ingest/v2/cli/cmds/sharepoint.py

```diff
@@ -0,0 +1,112 @@
+from unstructured.ingest.v2.processes.connectors.sharepoint import CONNECTOR_TYPE
+
+
+@dataclass
+class SharepointCliConnectionConfig(CliConfig):
+    @staticmethod
+    def get_cli_options() -> list[click.Option]:
+        options = [
+            click.Option(
+                ["--client-id"],
+                default=None,
+                type=str,
+                help="Sharepoint app client ID",
+            ),
+            click.Option(
+                ["--client-cred"],
+                default=None,
+                type=str,
+                help="Sharepoint app secret",
+            ),
+            click.Option(
+                ["--site"],
+                default=None,
+                type=str,
+                help="Sharepoint site url. Process either base url e.g \
+                    https://[tenant].sharepoint.com  or relative sites \
+                    https://[tenant].sharepoint.com/sites/<site_name>. \
+                    To process all sites within the tenant pass a site url as \
+                    https://[tenant]-admin.sharepoint.com.\
+                    This requires the app to be registered at a tenant level",
+            ),
+        ]
+        return options
+
+
+@dataclass
+class SharepointCliIndexerConfig(CliConfig):
+    @staticmethod
+    def get_cli_options() -> list[click.Option]:
+        options = [
+            click.Option(
+                ["--path"],
+                default=None,
+                type=str,
+                help="Path from which to start parsing files. If the connector is to \
+                process all sites within the tenant this filter will be applied to \
+                all sites document libraries.",
+            ),
+            click.Option(
+                ["--recursive"],
+                is_flag=True,
+                default=False,
+                help="Recursively download files in their respective folders "
+                "otherwise stop at the files in provided folder level.",
+            ),
+            click.Option(
+                ["--omit-files"],
+                is_flag=True,
+                default=False,
+                help="Don't process files.",
+            ),
+            click.Option(
+                ["--omit-pages"],
+                is_flag=True,
+                default=False,
+                help="Don't process site pages.",
+            ),
+            click.Option(
+                ["--permissions-application-id"],
+                type=str,
+                help="Microsoft Graph API application id",
+            ),
+            click.Option(
+                ["--permissions-client-cred"],
+                type=str,
+                help="Microsoft Graph API application credentials",
+            ),
+            click.Option(
+                ["--permissions-tenant"],
+                type=str,
+                help="e.g https://contoso.onmicrosoft.com to get permissions data within tenant.",
```

Comment:
In the [connector file](https://github.com/Unstructured-IO/unstructured/pull/3314/files#diff-a9f2bb5903c5d53c788b3cc4d4085c6571058951934d63d94301446923b02d6cR76), these are expected to be tied to the ConnectionConfig rather than the IndexerConfig